api/views/canteen.py

```diff
@@ -1064,6 +1064,24 @@ def annotate_actions(queryset, year):
         # prep add diag action
         diagnostics = Diagnostic.objects.filter(canteen=OuterRef("pk"), year=year)
         user_canteens = user_canteens.annotate(diagnostic_for_year=Subquery(diagnostics.values("id")))
+        is_central_cuisine_query = Q(production_type=Canteen.ProductionType.CENTRAL) | Q(
+            production_type=Canteen.ProductionType.CENTRAL_SERVING
+        )
+        is_satellite_query = Q(production_type=Canteen.ProductionType.ON_SITE_CENTRAL)
+        incomplete_canteen_data_query = (
+            Q(yearly_meal_count=None)
+            | Q(daily_meal_count=None)
+            | Q(siret=None)
+            | Q(name=None)
+            | Q(city_insee_code=None)
+            | Q(production_type=None)
+            | Q(management_type=None)
+            | (is_central_cuisine_query & Q(satellite_canteens_count=None))
+            | (is_satellite_query & Q(central_producer_siret=None))
+        )
+        # TODO: sector checks : https://stackoverflow.com/questions/42956736/querying-manytomany-in-django-giving-me-duplicate-results
+        # TODO: line ministry
```

Comment:
Side note : I wonder if querying the sector-canteen relation table instead of via the canteen will solve the problem you encountered?